src/metabase/task/delete_inactive_tables.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.task.delete-inactive-tables
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.schedule.cron :as cron]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [java-time :as t]
+   [metabase.config :as config]
+   [metabase.models.permissions :as perms]
+   [metabase.models.task-history :as task-history]
+   [metabase.task :as task]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private inactive-table-max-days
+  "The maximum number of days to keep inactive tables around for."
+  (or (config/config-int :inactive-table-max-days) 180))
+
+(def ^:private job-key     "metabase.task.delete-inactive-tables.job")
+(def ^:private trigger-key "metabase.task.delete-inactive-tables.trigger")
+
+(defn- delete-inactive-tables!
+  []
+  (log/debug (trs "Cleaning up Inactive Tables older than {0} days" inactive-table-max-days))
+  (task-history/with-task-history {:task "delete-inactive-tables"}
+    (t2/with-transaction [_conn]
+      (let [to-delete-tables (t2/select [:model/Table :id :db_id :schema]
+                                        :active false
+                                        :updated_at [:< (t/minus (t/zoned-date-time) (t/seconds inactive-table-max-days))])]
+        (when (seq to-delete-tables)
+          (t2/delete! :model/Table :id [:in (map :id to-delete-tables)])
+          (t2/delete! :model/Permissions
+                      {:where (into [:or]
+                                    (for [{:keys [id schema db_id]} to-delete-tables]
+                                      [:like :object (str "%" (perms/data-perms-path db_id schema id) "%")]))}))))))
```

Comment:
Doesn't this get triggered by the hook?

```
(t2/define-before-delete :model/Table
  [{:keys [db_id schema id]}]
  (t2/delete! Permissions :object [:like (str (perms/data-perms-path db_id schema id) "%")]))
  ```
  
  
  If not, I'd maybe extract the `[:like ...` logic into a helper function in `metabase.models.Table` that's used by both the before-delete hook and this task, just to make extra sure they stay in sync down the line

Reply:
oh I assumed it was done in the `after-delete` so I didn't know it's exists.

anw I think it should all be handled by table hook.

but I think there is a bug in `before-delete`, it doesn't clean up the new feature permissions like `/download/limited/db/3/schema/public/table/14/` or `/data-model/db/3/schema/public/table/14/`.

I've updated it to handle those permissions as well in: dcc27b0 

I'll also ask for review from Noah